pre-sync/oci-image-verification/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os/exec"
+	"sync"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	SourceURL    = "configsync.gke.io/source-url"
+	SourceCommit = "configsync.gke.io/source-commit"
+)
+
+var authorized bool
+var authMutex sync.Mutex // Add a mutex for thread safety
+
+// Function to extract annotations from JSON
+func getAnnotations(raw []byte) (map[string]string, error) {
+	var metadata map[string]interface{}
+	if err := json.Unmarshal(raw, &metadata); err != nil {
+		return nil, err
+	}
+	if annotations, ok := metadata["metadata"].(map[string]interface{})["annotations"].(map[string]interface{}); ok {
+		annotationsMap := make(map[string]string)
+		for k, v := range annotations {
+			annotationsMap[k] = fmt.Sprintf("%v", v)
+		}
+		return annotationsMap, nil
+	}
+	return nil, fmt.Errorf("no annotations found")
+}
+
+func validateImage(image string) error {
+	cmd := exec.Command("cosign", "verify", image, "--key", "/cosign-key/cosign.pub")
+	log.Printf("command %s", cmd.String())
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("cosign verification failed: %s, output: %s", err, string(output))
```

Comment:
Suggest adding image in the error if output doesn't already include that information. Users would like to know which image failed verification.

Reply:
The image info is included in the error returned by the ValidateImage function. Added image info to the success message so that user would see the passed ones in log.